From 245c16450ee402792fcc22aa3d849d87b1f70234 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@centricular.com>
Date: Fri, 18 Oct 2024 11:08:45 -0300
Subject: [PATCH 3/3] meson: Use a temporary directory instead of a temporary
 file for tests

Rust 1.82 has started to use an ArArchiveBuilder that, instead of
writing to the output, attempts to rename a temporary file. Since Python
has an open handle to reap it on scope exit, all calls with such paths
will always fail.
---
 meson/query-rustc.py | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/meson/query-rustc.py b/meson/query-rustc.py
index 2b169b1..aa55723 100755
--- a/meson/query-rustc.py
+++ b/meson/query-rustc.py
@@ -64,7 +64,6 @@ def retrive_version_info(output, query):
 
 if __name__ == "__main__":
     args = parser.parse_args()
-    dummy_out = tempfile.NamedTemporaryFile()
     query = args.query
     query_arg = None
     rustc_cmd = [Path(args.RUSTC).as_posix()]
@@ -89,13 +88,15 @@ if __name__ == "__main__":
     if query == 'native-static-libs':
         rustc_cmd.extend(['--crate-type', 'staticlib'])
         rustc_cmd.append(os.devnull)
-        rustc_cmd.extend(['-o', dummy_out.name])
+        rustc_cmd.extend(['-o', 'librsvg-query.a'])
 
-    query_results = subprocess.run(
-        rustc_cmd,
-        capture_output=True,
-        text=True,
-    )
+    with tempfile.TemporaryDirectory() as d:
+        query_results = subprocess.run(
+            rustc_cmd,
+            capture_output=True,
+            text=True,
+            cwd=d
+        )
     if query == 'native-static-libs':
         retrieve_native_static_libs_from_output(query_results.stderr)
     elif query == 'default-host-toolchain' or query == 'stable-actual-version':
-- 
2.44.0.windows.1

